### PR TITLE
feat:add new gpu index:GPU_BRUTE_FORCE and limit gpu index metric type

### PR DESCRIPTION
**Pull request by cqy123456**
issue: https://github.com/milvus-io/milvus/issues/29230
this pr do these things:
1. add gpu brute force;
2. limit gpu index only support l2 / ip;

### CODE REVIEW

**Comment by czs007** [comment]
rerun ut

**Comment by czs007** [comment]
@cqy123456  please run clang-format to fix the code check  issue

**Comment by cqy123456** [comment]
rerun ut

**Comment by cqy123456** [comment]
rerun ut

**Comment by cqy123456** [comment]
rerun ut

**Comment by cqy123456** [comment]
rerun ut

**Comment by cqy123456** [comment]
rerun ut

File: internal/core/src/config/ConfigKnowhere.cpp [hunk 1]
```diff
@@ -93,9 +93,14 @@ KnowhereInitGPUMemoryPool(const uint32_t init_size, const uint32_t max_size) {
     if (init_size == 0 && max_size == 0) {
         knowhere::KnowhereConfig::SetRaftMemPool();
         return;
+    } else if (init_size > max_size) {
+        PanicInfo(ConfigInvalid,
```
Comment (yah01) [inline internal/core/src/config/ConfigKnowhere.cpp:97]: try this:
```
PanicInfo(ConfigInvalid,
                  "Error Gpu memory pool params: init_size {} can't not large "
                  "than max_size {}.", init_size, max_size);
```

File: pkg/util/indexparamcheck/raft_ivf_pq_checker.go [hunk 2]
```diff
@@ -15,7 +15,9 @@ func (c *raftIVFPQChecker) CheckTrain(params map[string]string) error {
 	if err := c.ivfBaseChecker.CheckTrain(params); err != nil {
 		return err
 	}
-
+	if !CheckStrByValues(params, Metric, RaftMetrics) {
+		return fmt.Errorf("metric type not found or not supported, supported: %v", RaftMetrics)
```
Comment (yah01) [inline pkg/util/indexparamcheck/raft_ivf_pq_checker.go:19]: ->
```
return merr.WrapErrMetricNotFound(RaftMetrics)
```

**Review by yah01 — CHANGES_REQUESTED** [review]

File: pkg/util/indexparamcheck/raft_ivf_pq_checker_test.go [hunk 3]
```diff
@@ -123,7 +123,7 @@ func Test_raftIVFPQChecker_CheckTrain(t *testing.T) {
 		{validParamsMzero, true},
 		{p1, true},
 		{p2, true},
-		{p3, true},
+		{p3, false},
```
Comment (czs007) [inline pkg/util/indexparamcheck/raft_ivf_pq_checker_test.go:126]: why?

File: pkg/util/indexparamcheck/raft_ivf_pq_checker.go (hunk 2 above)
Comment (cqy123456) [reply pkg/util/indexparamcheck/raft_ivf_pq_checker.go:19]: import cycle

File: internal/core/src/config/ConfigKnowhere.cpp (hunk 1 above)
Comment (cqy123456) [reply internal/core/src/config/ConfigKnowhere.cpp:97]: updated

File: pkg/util/indexparamcheck/raft_ivf_pq_checker_test.go (hunk 3 above)
Comment (cqy123456) [reply pkg/util/indexparamcheck/raft_ivf_pq_checker_test.go:126]: GPU index not support cosine metric now

**Comment by cqy123456** [comment]
rerun ut

**Comment by samhuang-z** [comment]
rerun ut

**Pull request merged**